### DISCUSSION TITLE
GitInfo -> Devlooped.SponsorLink dependency violates privacy and collects developer PII without consent

### DISCUSSION THREAD

**Discussion by Eonasdan** [dotnet/maui#16642] [General]
Since @drasticactions closed my ticket and locked my ticket without actually reading what I posted. I'll post it here again with more clarity.

MAUI has a dependency on `GitInfo` which has a dependency on `Devlooped.SponsorLink`

`Devlooped.SponsorLink` is current a hot topic. It collects the git user email address and posts it to an Azure blob storage. And as this user points out in the Moq discussion, it adds a thread.sleep.

![image](https://github.com/dotnet/maui/assets/1006516/e9efb4ad-78bb-4dd8-a4d2-db31d53cd45b)

#16638

**Comment by drasticactions** [comment]
https://github.com/devlooped/GitInfo

AFAIK it (GitInfo) doesn't use it, but it is developed by the same people. It wouldn't matter if it did, because Moq is used for testing. It's code that doesn't end up in the build output for GitInfo itself (and so that doesn't end up as part of MAUI in its build process, nor your builds).

I closed it because I knew this. Again, you have many places to talk about it, this isn't one of them.

**Comment by hartez** [comment]
To clarify, the version of GitInfo used by the MAUI build process is an earlier version without the code in question. [More details](https://github.com/dotnet/maui/discussions/16629#discussioncomment-6682432)
